homeassistant/components/tplink_omada/switch.py

```diff
@@ -96,109 +170,82 @@ def _wan_connect_disconnect(
     )
 
 
-GATEWAY_PORT_SWITCHES: list[GatewayPortSwitchEntityDescription] = [
-    GatewayPortSwitchEntityDescription(
+SWITCH_PORT_DETAILS_SWITCHES: list[OmadaSwitchPortSwitchEntityDescription] = [
+    OmadaSwitchPortSwitchEntityDescription(
+        key="poe",
+        translation_key="poe_control",
+        exists_func=lambda d, p: d.device_capabilities.supports_poe
+        and p.type != PortType.SFP,
+        set_func=lambda client, device, port, enable: client.update_switch_port(
+            device, port, overrides=SwitchPortOverrides(enable_poe=enable)
+        ),
+        update_func=lambda p: p.poe_mode != PoEMode.DISABLED,
+        entity_category=EntityCategory.CONFIG,
+    )
+]
+
+GATEWAY_PORT_STATUS_SWITCHES: list[OmadaGatewayPortStatusSwitchEntityDescription] = [
+    OmadaGatewayPortStatusSwitchEntityDescription(
         key="wan_connect_ipv4",
         translation_key="wan_connect_ipv4",
-        exists_func=lambda p: p.mode == GatewayPortMode.WAN,
+        exists_func=lambda _, p: p.mode == GatewayPortMode.WAN,
         set_func=partial(_wan_connect_disconnect, ipv6=False),
         update_func=lambda p: p.wan_connected,
+        refresh_after_set=True,
     ),
-    GatewayPortSwitchEntityDescription(
+    OmadaGatewayPortStatusSwitchEntityDescription(
         key="wan_connect_ipv6",
         translation_key="wan_connect_ipv6",
-        exists_func=lambda p: p.mode == GatewayPortMode.WAN and p.wan_ipv6_enabled,
+        exists_func=lambda _, p: p.mode == GatewayPortMode.WAN and p.wan_ipv6_enabled,
         set_func=partial(_wan_connect_disconnect, ipv6=True),
         update_func=lambda p: p.ipv6_wan_connected,
+        refresh_after_set=True,
     ),
 ]
 
-
-def get_port_base_name(port: OmadaSwitchPortDetails) -> str:
-    """Get display name for a switch port."""
-
-    if port.name == f"Port{port.port}":
-        return f"{port.port}"
-    return f"{port.port} ({port.name})"
+GATEWAY_PORT_CONFIG_SWITCHES: list[OmadaGatewayPortConfigSwitchEntityDescription] = [
+    OmadaGatewayPortConfigSwitchEntityDescription(
+        key="poe",
+        translation_key="poe_control",
+        exists_func=lambda _, port: port.poe_mode != PoEMode.NONE,
+        set_func=lambda client, device, port, enable: client.set_gateway_port_settings(
+            port.port_number, GatewayPortSettings(enable_poe=enable), device
+        ),
+        update_func=lambda p: p.poe_mode == PoEMode.ENABLED,
```

Comment:
Why is the update function here different from the one in line 182? A user expects that the switch behaves the same on the gateway and the switch

Reply:
It's a fair point. There is reason why one is written as `poe_mode != PoEMode.DISABLED` and the other is written as `poe_mode == PoEMode.ENABLED`, but not a great reason. For switches, there are three possible options - DISABLED, ENABLED and DEVICE_SETTINGS. We treat DEVICE_SETTINGS as On. The gateway only has ON and OFF as options. Practically, there is no difference in operation - switching the switch off turns off PoE in both cases.

Reply:
Can we use for both `poe_mode != PoEMode.DISABLED`. It looks more consistent

```suggestion
        update_func=lambda p: p.poe_mode  != PoEMode.DISABLED,
```